src/BRSRC13/CORE/MATH/vector.c

```diff
@@ -2,121 +2,163 @@
 void BrVector2Accumulate(br_vector2* v1, br_vector2* v2) {
     LOG_TRACE("(%p, %p)", v1, v2);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] += v2->v[0];
+    v1->v[1] += v2->v[1];
 }
 
 // IDA: void __cdecl BrVector2Sub(br_vector2 *v1, br_vector2 *v2, br_vector2 *v3)
 void BrVector2Sub(br_vector2* v1, br_vector2* v2, br_vector2* v3) {
     LOG_TRACE("(%p, %p, %p)", v1, v2, v3);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = v2->v[0] - v3->v[0];
+    v1->v[1] = v2->v[1] - v3->v[1];
 }
 
 // IDA: void __cdecl BrVector2Scale(br_vector2 *v1, br_vector2 *v2, br_scalar s)
 void BrVector2Scale(br_vector2* v1, br_vector2* v2, br_scalar s) {
     LOG_TRACE("(%p, %p, %f)", v1, v2, s);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = v2->v[0] * s;
+    v1->v[1] = v2->v[1] * s;
 }
 
 // IDA: void __cdecl BrVector2InvScale(br_vector2 *v1, br_vector2 *v2, br_scalar s)
 void BrVector2InvScale(br_vector2* v1, br_vector2* v2, br_scalar s) {
     LOG_TRACE("(%p, %p, %f)", v1, v2, s);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = v2->v[0] / s;
+    v1->v[1] = v2->v[1] / s;
 }
 
 // IDA: br_scalar __cdecl BrVector2Dot(br_vector2 *v1, br_vector2 *v2)
 br_scalar BrVector2Dot(br_vector2* v1, br_vector2* v2) {
     LOG_TRACE("(%p, %p)", v1, v2);
-    NOT_IMPLEMENTED();
+
+    return BR_MAC2(v1->v[0], v2->v[0], v1->v[1], v2->v[1]);
 }
 
 // IDA: br_scalar __cdecl BrVector2Length(br_vector2 *v1)
 br_scalar BrVector2Length(br_vector2* v1) {
     LOG_TRACE("(%p)", v1);
-    NOT_IMPLEMENTED();
+
+    return sqrtf(BR_MAC2(v1->v[0], v1->v[0], v1->v[1], v1->v[1]));
 }
 
 // IDA: br_scalar __cdecl BrVector2LengthSquared(br_vector2 *v1)
 br_scalar BrVector2LengthSquared(br_vector2* v1) {
     LOG_TRACE("(%p)", v1);
-    NOT_IMPLEMENTED();
+
+    return BR_MAC2(v1->v[0], v1->v[0], v1->v[1], v1->v[1]);
 }
 
 // IDA: void __cdecl BrVector3Copy(br_vector3 *v1, br_vector3 *v2)
 void BrVector3Copy(br_vector3* v1, br_vector3* v2) {
     LOG_TRACE("(%p, %p)", v1, v2);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = v2->v[0];
+    v1->v[1] = v2->v[1];
+    v1->v[2] = v2->v[2];
 }
 
 // IDA: void __cdecl BrVector3Set(br_vector3 *v1, br_scalar s1, br_scalar s2, br_scalar s3)
 void BrVector3Set(br_vector3* v1, br_scalar s1, br_scalar s2, br_scalar s3) {
     LOG_TRACE("(%p, %f, %f, %f)", v1, s1, s2, s3);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = s1;
+    v1->v[1] = s2;
+    v1->v[2] = s3;
 }
 
 // IDA: void __cdecl BrVector3SetInt(br_vector3 *v1, int i1, int i2, int i3)
 void BrVector3SetInt(br_vector3* v1, int i1, int i2, int i3) {
     LOG_TRACE("(%p, %d, %d, %d)", v1, i1, i2, i3);
-    NOT_IMPLEMENTED();
+
+    v1->v[0] = BR_SCALAR(i1);
+    v1->v[1] = BR_SCALAR(i2);
+    v1->v[2] = BR_SCALAR(i3);
 }
 
 // IDA: void __cdecl BrVector3SetFloat(br_vector3 *v1, float f1, float f2, float f3)
 void BrVector3SetFloat(br_vector3* v1, float f1, float f2, float f3) {
     LOG_TRACE8("(%p, %f, %f, %f)", v1, f1, f2, f3);
 
-    v1->v[0] = f1;
-    v1->v[1] = f2;
-    v1->v[2] = f3;
+    v1->v[0] = BR_SCALAR(f1);
```

Comment:
where does the wrapping to `BR_SCALAR(f1)` come from? Is it necessary? Are you interested in adding support for running brender in `fixed` mode?